apps/arena/lib/arena/configuration.ex

```diff
@@ -13,29 +13,35 @@ defmodule Arena.Configuration do
       |> File.read()
 
     config = Jason.decode!(config_json, [{:keys, :atoms}])
-    characters = parse_characters_config(get_characters_config())
     client_config = get_client_config()
-    game_config = get_game_configuration()
-    map_config = parse_map_config(get_map_config())
-    items_config = get_consumable_items_configuration()
 
     config
-    |> Map.put(:characters, characters)
-    |> Map.put(:game, game_config)
-    |> Map.put(:map, map_config)
-    |> Map.put(:characters, characters)
-    |> Map.put(:items, items_config)
+    |> Map.merge(get_current_game_configuration())
     |> Map.put(:client_config, client_config)
   end
 
-  defp get_map_config() do
+  defp get_current_game_configuration do
     gateway_url = Application.get_env(:arena, :gateway_url)
 
     {:ok, payload} =
-      Finch.build(:get, "#{gateway_url}/curse/configuration/map", [{"content-type", "application/json"}])
+      Finch.build(:get, "#{gateway_url}/curse/configuration/current", [{"content-type", "application/json"}])
       |> Finch.request(Arena.Finch)
 
     Jason.decode!(payload.body, [{:keys, :atoms}])
+    |> Map.update!(:map, fn maps ->
```

Comment:
I'm not sure why I have to do this, maybe I'm missing some kind of option for nested jsons

Reply:
Yes and no, its impossible for Jason to differentiate between a string and a string representing a JSON, that's why you need to manually decode each field

The actual problem is likely in the server response and how it is encoding things